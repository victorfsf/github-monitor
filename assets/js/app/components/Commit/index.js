import Commit from './Commit';

export default Commit;
