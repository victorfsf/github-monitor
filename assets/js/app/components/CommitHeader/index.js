import CommitHeader from './CommitHeader';

export default CommitHeader;
