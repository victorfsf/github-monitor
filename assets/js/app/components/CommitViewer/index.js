import CommitViewer from './CommitViewer';

export default CommitViewer;
