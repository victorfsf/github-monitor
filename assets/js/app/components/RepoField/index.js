import RepoField from './RepoField';

export default RepoField;
