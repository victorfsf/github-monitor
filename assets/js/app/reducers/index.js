import { combineReducers } from 'redux';

// TODO: write actual reducers
const mockedReducer = (state, action) => ({ mock: action });

const rootReducer = combineReducers({
  mockedReducer,
});

export default rootReducer;
