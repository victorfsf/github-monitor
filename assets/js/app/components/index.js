import Box from './Box';
import CommitViewer from './CommitViewer';
import Container from './Container';
import LoginForm from './LoginForm';
import RepoField from './RepoField';
import Spinner from './Spinner';
import Pagination from './Pagination';
import NotFound from './NotFound';

export {
  Box,
  CommitViewer,
  Container,
  LoginForm,
  RepoField,
  Spinner,
  Pagination,
  NotFound,
};
