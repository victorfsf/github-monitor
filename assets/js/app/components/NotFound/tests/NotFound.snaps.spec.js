import React from 'react';
import renderer from 'react-test-renderer';
import { NotFound } from 'app/components';
import * as ReactRouter from 'react-router-dom';

ReactRouter.Link = jest.fn(props => (
  <a href={props.to}>{props.children}</a>
));


describe('NotFound', () => {
  let Component;
  let tree;

  test('does not render without all props', () => {
    global.console = {
      error: jest.fn(),
    };
    renderer.create(<NotFound />);
    renderer.create(<NotFound title="title" />);
    renderer.create(<NotFound title="title" faClassName="fa" />);
    renderer.create(<NotFound title="title">Child</NotFound>);
    expect(global.console.error).toHaveBeenCalledTimes(3);
  });

  test('renders', () => {
    Component = renderer.create(
      <NotFound title="title" faClassName="fa fa-test">
        <div>Node</div>
      </NotFound>,
    );
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
