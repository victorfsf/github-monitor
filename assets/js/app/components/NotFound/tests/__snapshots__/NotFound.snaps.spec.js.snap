// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders 1`] = `
<div
  className="justify-content-center"
>
  <div
    className="not-found"
  >
    <i
      className="fa fa-test"
    />
    <div
      className="title"
    >
      title
    </div>
    <div
      className="description"
    >
      <div>
        Node
      </div>
    </div>
  </div>
</div>
`;
